aws/resource_aws_default_network_acl.go

```diff
@@ -64,32 +68,41 @@ func resourceAwsDefaultNetworkAcl() *schema.Resource {
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"from_port": {
-							Type:     schema.TypeInt,
-							Required: true,
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IsPortNumberOrZero,
 						},
 						"to_port": {
-							Type:     schema.TypeInt,
-							Required: true,
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IsPortNumberOrZero,
 						},
 						"rule_no": {
-							Type:     schema.TypeInt,
-							Required: true,
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(1, 32766),
 						},
 						"action": {
 							Type:     schema.TypeString,
 							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								ec2.RuleActionAllow,
+								ec2.RuleActionDeny,
+							}, true),
 						},
 						"protocol": {
 							Type:     schema.TypeString,
 							Required: true,
 						},
 						"cidr_block": {
-							Type:     schema.TypeString,
-							Optional: true,
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.IsCIDR,
```

Comment:
I think we need to allow empty strings in the cidr_block/ipv6_cidr_block attributes for Terraform 0.11 and earlier.

Reply:
done